test/shared/src/main/scala/zio/test/mock/MockException.scala

```diff
@@ -0,0 +1,15 @@
+package zio.test.mock
+
+import zio.test.{ AssertResult, Assertion }
+
+sealed trait MockException extends Throwable
+
+final case class PredicateFailedException(predicate: AssertResult) extends MockException
```

Comment:
Can you place these two subtypes inside a companion object for `MockException`? That's the standard ZIO pattern (and a common Scala idiom).